File: backend/crypto-tracker/src/main/java/com/demo/cryptotracker/model/Cryptocurrency.java
package com.demo.cryptotracker.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotNull;
import javax.persistence.Entity;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Cryptocurrency extends ModelEntity<Long> {

    @NotNull
    private String symbol;

    @NotNull
    private String name;

    private String hashingAlgorithm;
    private String categories;
    private String description;
    private String programmingLanguageImplementation;
    private String image;
    private String links;
    private Long releaseYear;
}

File: frontend/crypto-tracker/src/app/model/cryptocurrency.ts
export class Cryptocurrency {

    id: number;
    symbol: string;
    name: string;
    hashingAlgorithm: string;
    categories: string; 
    description: string;
    links: string;
    image: string;
    programmingLanguageImplementation: string;
    releaseYear: number;

    constructor(id: number, symbol: string, name: string, hashingAlgorithm: string, categories: string, description: string, links: string,
                image: string, programmingLanguageImplementation: string, releaseYear: number) {
        this.id = id;
        this.symbol = symbol;
        this.name = name;
        this.hashingAlgorithm = hashingAlgorithm;
        this.categories = categories;
        this.description = description;
        this.links = links;
        this.image = image;
        this.programmingLanguageImplementation = programmingLanguageImplementation;
        this.releaseYear = releaseYear;
    }
 }